### PR TITLE
fdt: add utility functions

### PR DESCRIPTION
**Pull request by rminnich**
These functions are needed to enable kexec'ing a
Flattened Image Tree.

Signed-off-by: Ronald G Minnich <rminnich@gmail.com>

### CODE REVIEW

File: pkg/dt/fdt.go [hunk 1]
```diff
@@ -411,3 +413,35 @@ func (fdt *FDT) Write(f io.Writer) (int, error) {
...
+	return fdt.RootNode.Find(func(n *Node) bool {
+		return n.Name == name
+	})
+}
+
+// Bytes returns a []byte for the "data" property of a node,
+// or an error if the node has no data property.
+func (fdt *FDT) Bytes(name string) ([]byte, error) {
+	n, ok := fdt.NodeByName(name)
+	log.Printf("Bytes: nodebyname for %q is %s", name, fdt)
```
Comment (rjoleary) [inline pkg/dt/fdt.go:428]: Make this line verbose, otherwise it will be super noisy.

File: pkg/dt/fdt.go [hunk 2]
```diff
@@ -411,3 +413,35 @@ func (fdt *FDT) Write(f io.Writer) (int, error) {
...
+	if !ok {
+		return nil, os.ErrExist
```
Comment (rjoleary) [inline pkg/dt/fdt.go:430]: Can you instead do `fmt.Errorf("could not find device tree node %q", name)`

File: pkg/dt/fdt.go [hunk 3]
```diff
@@ -411,3 +413,35 @@ func (fdt *FDT) Write(f io.Writer) (int, error) {
...
+// NodeByName finds a node by name.
+func (fdt *FDT) NodeByName(name string) (*Node, bool) {
+	return fdt.RootNode.Find(func(n *Node) bool {
+		return n.Name == name
+	})
+}
+
+// Bytes returns a []byte for the "data" property of a node,
+// or an error if the node has no data property.
+func (fdt *FDT) Bytes(name string) ([]byte, error) {
```
Comment (rjoleary) [inline pkg/dt/fdt.go:426]: It looks like this is specific to FIT, but not Device Tree in general. Perhaps make it a utility in a separate package called `fit` or rename to `FitBytes`.

File: pkg/dt/fdt.go [hunk 4]
```diff
@@ -411,3 +413,35 @@ func (fdt *FDT) Write(f io.Writer) (int, error) {
...
+	p, ok := n.LookProperty("data")
+	if !ok {
+		return nil, fmt.Errorf("%s does not have a data property", n)
+	}
+	return p.Value, nil
+}
+
+// Reader returns a Reader for the data property of a Node with name 'name',
+// or an error if the node is not found or the node has no data property
+func (fdt *FDT) Reader(name string) (io.ReaderAt, error) {
```
Comment (rjoleary) [inline pkg/dt/fdt.go:441]: Same as previous comment. Move to `fit` package or call `FitBytesReader`.

File: pkg/dt/fdt.go (hunk 3 above)
Comment (rminnich) [reply pkg/dt/fdt.go:426]: done

Comment (rminnich) [reply pkg/dt/fdt.go:426]: not really. IIRC data is a comment property for a node?

File: pkg/dt/fdt.go (hunk 4 above)
Comment (rminnich) [reply pkg/dt/fdt.go:441]: same argument. data tickles my memory as being a common property name?

File: pkg/dt/fdt.go (hunk 2 above)
Comment (rminnich) [reply pkg/dt/fdt.go:430]: done

**Review by rjoleary — APPROVED** [review]

**Review by rjoleary — APPROVED** [review]

**Pull request merged**